mantidproject/mantid
# Indirect -  IqtFitSequential algorithm

Added the `IqtFitSequential` algorithm to replace the call to the IndirectDataAnalysis script in the IqtFit interface and replace the functionality of fury_fit_sequential

**To test:**
- `iris26176_graphite002_iqt.nxs` can be found in the unit test data
- Running this script should ensure the algorithm functionality  

```
function = r'name=LinearBackground,A0=0,A1=0,ties=(A1=0);name=UserFunction,Formula=Intensity*exp(-(x/Tau)),Intensity=1,Tau=0.0247558;ties=(f1.Intensity=1-f0.A0)'
iqt_ws = Load(Filename='iris26176_graphite002_iqt.nxs',
                            OutputWorkspace='iris26176_graphite002_iqt')
result, params, fit_group = IqtFitSequential(InputWorkspace=iqt_ws,
                                                     Function=function,
                                                     FitType='1E_s',
                                                     StartX=0,
                                                     EndX=0.24,
                                                     SpecMin=0,
                                                     SpecMax=16,
                                                     ConstrainIntensities=True)
```
- The interface (Interfaces > Indirect > Data Analysis > `Iqt(Fit)`) should also be checked to ensure the it is updating properly and running the correct algorithms.
- By changing the Fit Type in the interface and running the interface using the same data set from the above script, you should see the mini plot updated each time with both the difference and corrected curves (red and blue)

Fixes #15440.

**Updates to the interface in this issue should also fix #15343 and therefore this PR can close both issues**

[RELEASE NOTES](https://github.com/mantidproject/mantid/blob/15440_IqtFitSequential_algorithm/docs/source/release/v3.7.0/indirect_inelastic.rst#algorithms)

---
#### Reviewer

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):
##### Code Review
##### Functional Tests
- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
